src/ledger/NetworkConfig.h

```diff
@@ -140,6 +149,11 @@ class SorobanNetworkConfig
     // Fee for propagating 1KB of data
     int64_t feePropagateData1KB() const;
 
+#ifdef BUILD_TESTS
+    uint32_t& maxContractDataKeySizeBytes();
```

Comment:
I'm not sure this + `setSorobanNetworkConfig` are necessary  - it seems easy to create entries/keys that exceed the limits (just use the `Bytes` `ScVal` and resize it to the necessary limit - you can also use the initial value as well).

Reply:
I would like to keep this for now if that's okay. 
It's nice to be able to override the config for testing. For example, I was going to add a contract function that writes a large `Bytes` entry and making sure that entry (instead of the contract code itself) triggers the limit. This wouldn't be easy to do without being able to override the config.

Reply:
Ok, let's leave this for now, but in general I'd like to use something a bit less specific for overriding the configs.